metadata-ingestion/examples/library/add_lineage_dataset_to_dataset_with_query_node.py

```diff
@@ -0,0 +1,28 @@
+from datahub.metadata.urns import DatasetUrn
+from datahub.sdk.main_client import DataHubClient
+
+client = DataHubClient.from_env()
+
+upstream_urn = DatasetUrn(platform="snowflake", name="upstream_table")
+downstream_urn = DatasetUrn(platform="snowflake", name="downstream_table")
+
+query_text = """
+from pyspark.sql import SparkSession
+
+spark = SparkSession.builder.appName("HighValueFilter").getOrCreate()
+df = spark.read.table("customers")
+high_value = df.filter("lifetime_value > 10000")
+high_value.write.saveAsTable("high_value_customers")
+"""
+
+client.lineage.add_lineage(
+    upstream=upstream_urn,
+    downstream=downstream_urn,
+    query_text=query_text,
```

Comment:
maybe we should change this to `transformation_logic` or something so that it's more clear that it's not a SQL query